src/App.tsx

```diff
@@ -1,39 +1,77 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Components/Clock/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component< {}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({
+        clockName: getRandomName(),
+      });
+    }, 3300);
+
+    document.addEventListener('contextmenu', this.righClickHandler);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    document.addEventListener('click', this.leftClickHandler);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(_: {}, prevState: State) {
+    const { clockName, hasClock } = this.state;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    if ((clockName !== prevState.clockName) && hasClock) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevState.clockName} to ${clockName}`);
+    }
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+    document.removeEventListener('contextmenu', this.righClickHandler);
+  }
 
-        {' time is '}
+  righClickHandler = (event: MouseEvent) => {
+    event.preventDefault(); // not to show the context menu
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    this.setState({
+      hasClock: false,
+    });
+  };
+
+  leftClickHandler = () => {
+    this.setState({
+      hasClock: true,
+    });
+  };
+
+  render() {
+    const {
+      clockName,
+      hasClock,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {hasClock && <Clock clockName={clockName} />}
```

Comment:
Just a suggestion, but I prefer this way of rendering components conditionally. Looks clearer to me
```suggestion
        {hasClock && (
          <Clock clockName={clockName} />
        )}
```